python/cudf/cudf/core/dataframe.py

```diff
@@ -3519,6 +3519,57 @@ def notnull(self):
         """
         return self.notna()
 
+    # `idxmin` and `idxmax` are stubbed here for writing expected
+    # tests. After feedback from the etl team on the libcudf API
+    # they will be replaced.
+    def idxmin(self, axis=0, skipna=True):
+        """
+        Return index of first occurrence of minimum over requested axis.
+        NA/null values are excluded.
+
+        Parameters:
+            axis : {0 or ‘index’, 1 or ‘columns’}, default 0
+                0 or ‘index’ for row-wise, 1 or ‘columns’ for column-wise
+            skipna : boolean, default True
+                Exclude NA/null values. If an entire row/column is NA,
+                the result will be NA.
+
+        Returns:
+            Series
+                Indexes of minima along the specified axis.
+
+        Raises:
+            ValueError
+                If the row/column is empty
+        """
+        # Returning pandas.Series.idxmax value until libcudf API
+        # requirements established.
+        return self.to_pandas().idxmin(axis=axis, skipna=skipna)
+
+    def idxmax(self, axis=0, skipna=True):
+        """
+        Return index of first occurrence of maximum over requested axis.
+        NA/null values are excluded.
+
+        Parameters:
+            axis : {0 or ‘index’, 1 or ‘columns’}, default 0
```

Comment:
Let's downscope for now and focus on only doing a column-wise `idxmax` for now. Let's update these docs accordingly to indicate axis=0 is not yet supported.

Reply:
`axis=0` would require all columns to be the same dtype, and we'd need a implementation that implicitly transposes the table as if it were a matrix. Doable, but an entirely different scenario.